include/kllvm/bindings/core/core.h

```diff
@@ -37,6 +37,9 @@ std::shared_ptr<kllvm::KOREPattern> simplify(
 bool is_sort_kitem(std::shared_ptr<kllvm::KORESort> const &sort);
 bool is_sort_k(std::shared_ptr<kllvm::KORESort> const &sort);
 
+std::shared_ptr<KOREPattern>
+evaluate_function(std::shared_ptr<KORECompositePattern> term);
```

Comment:
Btw, what's the difference between this function and `simplify_pattern`? Is this just a special case thereof?

Reply:
It's a special case; simplification works also for terms like `c(f())` where `c` is a constructor - this API is intended _only_ to directly evaluate functions and so it skips a couple of checks in the backend to get directly to the right code path.